core/pkg/artifacts/downloader.go

```diff
@@ -86,6 +89,126 @@ func (ad *ArtifactDownloader) getArtifactManifest(artifactID string) (manifest M
 	return manifest, nil
 }
 
+// getBatchEntriesWithFileUrls fetches file urls for the entries in
+// entriesToFetch and returns a list of the formatted manifest entries to
+// download in the current batch.
+func (ad *ArtifactDownloader) getBatchEntriesWithFileUrls(
+	artifactID string,
+	entriesToFetch []gql.ArtifactManifestEntryInput,
+	manifest Manifest,
+	manifestEntries map[string]ManifestEntry,
+) ([]ManifestEntry, error) {
+	var entries []ManifestEntry
+	response, err := gql.ArtifactFileURLsByManifestEntries(
+		ad.Ctx,
+		ad.GraphqlClient,
+		artifactID,
+		entriesToFetch,
+		manifest.StoragePolicyConfig.StorageLayout,
+		strconv.Itoa(int(manifest.Version)),
+		nil,
+	)
+	if err != nil {
+		return nil, err
+	}
+	for _, edge := range response.GetArtifact().GetFilesByManifestEntries().Edges {
```

Comment:
is it fair to assume that all the getter will return not nil value?

Reply:
I think so, unless the query returns an error, it should have all these fields